### PR TITLE
Fix PI0 embed_tokens.weight loading + WandB list metric logging

### LINKED ISSUE #3109: PI0 WARNING: Vision embedding key might need handling && WandB logging of key "loss_per_dim" was ignored
### Ticket Type

🐛 Bug Report (Something isn't working)

### Environment & System Info

```Shell
lerobot=0.4.4
```

### Description

Train PI0 as turorial.

Logs as follows: 

```bash
WARNING 2026-03-08 14:47:13 ing_pi0.py:1121 Vision embedding key might need handling: paligemma_with_expert.paligemma.model.vision_tower.visi
on_model.embeddings.patch_embedding.bias                                                                                                     
WARNING 2026-03-08 14:47:13 ing_pi0.py:1121 Vision embedding key might need handling: paligemma_with_expert.paligemma.model.vision_tower.visi
on_model.embeddings.patch_embedding.weight                                                                                                   
Remapped 777 state dict keys                                                                                                                 
Warning: Could not remap state dict keys: Error(s) in loading state_dict for PI0Policy:                                                      
        Missing key(s) in state_dict: "model.paligemma_with_expert.paligemma.model.language_model.embed_tokens.weight".                      

...

INFO 2026-03-08 14:59:15 ot_train.py:435 step:200 smpl:2K ep:5 epch:0.16 loss:0.375 grdn:15.012 lr:2.5e-06 updt_s:3.533 data_s:0.057
WARNING 2026-03-08 14:59:15 db_utils.py:181 WandB logging of key "loss_per_dim" was ignored as its type "<class 'list'>" is not handled by this wrapper.
```

There is the same question in PI05 https://github.com/huggingface/lerobot/issues/2307#issuecomment-3442314378, but no one propose it in PI0 pipeline.

How soon it will be solved?

### Context & Reproduction

_No response_

### Relevant logs or stack trace

```Shell

```

### Checklist

- [x] I have searched existing tickets to ensure this isn't a duplicate.
- [ ] I am using the latest version of the `main` branch.
- [x] I have verified this is not an environment-specific problem.

### Additional Info / Workarounds

_No response_

### PR DESCRIPTION
**Pull request by he-yufeng**
## What this PR does

Fixes two issues reported in #3109:

1. **PI0 state dict `embed_tokens.weight` missing after loading**: `_fix_pytorch_state_dict_keys` only creates this key from `lm_head.weight` (weight tying), but some checkpoints store `embed_tokens.weight` directly. Added explicit handling for this key.

2. **WandB silently drops `loss_per_dim`**: the logging wrapper only handles scalars. List/tuple metrics like `loss_per_dim` are now expanded into per-index entries (`loss_per_dim_0`, `loss_per_dim_1`, ...).

Fixes #3109

### CODE REVIEW

**Comment by he-yufeng** [comment]
Also bumping this one — the embed_tokens fix and WandB list metric issue are both still present in main.

**Comment by he-yufeng** [comment]
Rebased this branch onto current main and tightened the WandB list-metric fix so expanded metrics preserve the original step/custom-step semantics instead of letting WandB advance steps implicitly. I also added regression tests for both fixes: list/tuple metric expansion and direct PI0 embed_tokens.weight state-dict keys. Validation: ruff check on touched source/tests passed; ruff format --check on touched source/tests passed; py_compile on touched source/tests passed; git diff --check upstream/main..HEAD passed; direct behavior checks for WandB step/custom-step logging and PI0 embed_tokens key normalization passed. Local pytest note: running the new pytest files in this Windows venv currently times out during LeRobot's shared conftest/import initialization after loading torchvision; the new tests themselves are lightweight and should run under the project CI environment.

**Comment by imstevenpmwork** [comment]
Hello @he-yufeng 

Could you confirm this is still an issue in `0.6.0` ?
